lib/mayaUsd/fileio/shading/shadingModeUseRegistry.cpp

```diff
@@ -283,8 +285,8 @@ class UseRegistryShadingModeExporter : public UsdMayaShadingModeExporter
                 }
 
                 bool allowedNode = false;
```

Comment:
Using a MObjectHandleUnorderedSet has powerful benefits:

```suggestion
                if (!allowedNodes.count(dstPlug.node()) {
                    continue
                }
```

This replaces the whole `for` loop, and all `allowedNode` boolean logic.